lib/workingsetcache/cache.go

```diff
@@ -326,6 +339,8 @@ func (c *Cache) cacheSizeWatcher() {
 	updateCacheStatsHistory(&c.csHistory, &cs)
 	prev.Reset()
 	c.mu.Unlock()
+
+	goto again
```

Comment:
would it make sense to wrap the switching process in a func instead?
```
func (c *Cache) cacheSizeWatcher() {
	checkInterval := timeutil.AddJitterToDuration(time.Millisecond * 1500)
	t := time.NewTicker(checkInterval)
	defer t.Stop()

	var maxBytesSize uint64
	for {
		select {
		case <-c.stopCh:
			return
		case <-t.C:
		}
		if c.mode.Load() != split {
			continue
		}
		var cs fastcache.Stats
		curr := c.curr.Load()
		curr.UpdateStats(&cs)
		if cs.BytesSize >= uint64(0.9*float64(cs.MaxBytesSize)) {
			c.switchToWhole(maxBytesSize)
		}
	}
}
```

this would help getting rid of gotos, because goto in the end of the function is likely can be avoided